### DISCUSSION TITLE
Configure tags on VMs and other resources in a HCIBox deployment

### DISCUSSION THREAD

**Discussion by shishirm-cohesity** [microsoft/azure_arc#2540] [Q&A]
Very basic question but is there a way to specify tags that can be applied to resources created by the HCIBox deployment scripts? A quick search didn't reveal anything and my Azure knowledge is fairly rudimentary so it's tricky to understand if the scripts can be modified to support this.

**Comment by dkirby-ms** [comment]
Yes they could be modified to support this. You could do this in several ways but one easy way would be to add the tags using Az CLI after the deployment finishes. This way you dont need to modify the existing code, just query your deployment using Az CLI to get the resource identifiers, then use Az CLI again to add tags to resources as needed.

```
resource=$(az resource show -g demoGroup -n demostorage --resource-type Microsoft.Storage/storageAccounts --query "id" --output tsv)
az tag create --resource-id $resource --tags Dept=Finance Status=Normal
```

**Reply by shishirm-cohesity** [reply]
Thanks Dale. While this is certainly an option, it didn't quite work for us since there was some automation configured in our account to periodically remove assets that weren't tagged properly. Since the HCIBox deployment takes a while, the clean up was preventing the resources from being up long enough for us to alter their properties. What we eventually ended up doing was locating where tags were being used in HCIBox already, and add our required tags to that list. That way, when the resources came up, they were already tagged.

It might be worth considering an enhancement to the HCIBox configuration files or scripts to have a placeholder for user tags. That way, if folks need to use their own tags, then those can be injected into the creation process itself, instead of requiring a post-installation intervention.
